### ISSUE TITLE
Commitlint fails after squash merge

### ISSUE THREAD

**Issue by aj3sh** [opensource-nepal/commitlint#67]
**Describe the bug**
When a PR is created with a commit message of around 72 characters the commitlint action will pass. But when the PR is Squashed and Merged Github appends the PR number in the commit message. The appended commit message fails the commitlint after the PR is merged.
https://github.com/opensource-nepal/node-nepali-datetime/actions/runs/11974343317
<img width="599" alt="Screenshot 2024-11-22 at 8 42 35 PM" src="https://github.com/user-attachments/assets/97c34c6f-cdbe-4c38-b265-e4280c3a6a4a">

**To Reproduce**
Pre-requisite: Github Action is set up on the target branch with `pull_request` and `push` events.
1. Create a PR with a commit message of 72 characters.
2. The commitlint will pass.
3. Squash and merge the PR.
4. Leave the commit message GitHub suggests (PR number appended to the commit message).
5. The commitlint fails after the PR is merged as the commit message length exceeds 72 characters.

**Expected behavior**
The commitlint should be smart enough to overcome this situation.
Possible Solution:
- Ignore the appended PR number; common formats `... (#PR)` and `... #PR`.

**Environment**
-   Python version: N/A
-   Package version: v1.3.0
-   Operating system: Github Actions
-   Any other relevant information about your environment.

**Comment by kiyoon** [comment]
I think it should have an option to bypass the 72-character limit, and even configure the length. The length limit is not even on the conventional commit spec.

**Comment by aj3sh** [comment]
> I think it should have an option to bypass the 72-character limit, and even configure the length. The length limit is not even on the conventional commit spec.

True, the length check should be optional and limit should be configurable.

**Comment by SiegeSailor** [comment]
Hello @aj3sh, I ended up with something like below in my `releaserc.toml` for PSR (see https://github.com/python-semantic-release/python-semantic-release/issues/187#issuecomment-3505999740 for details):

```toml
[semantic_release]
commit_message = """\
chore: release {version}

Automatically generated by python-semantic-release
"""
```

I am not sure how a custom ignore pattern works here. Let me dive deeper into the whole commit linting and automation ecosystem and come back later.

**Comment by aj3sh** [comment]
> Hello @aj3sh, I ended up with something like below in my `releaserc.toml` for PSR (see https://github.com/python-semantic-release/python-semantic-release/issues/187#issuecomment-3505999740 for details):
> 
> ```toml
> [semantic_release]
> commit_message = """\
> chore: release {version}
> 
> Automatically generated by python-semantic-release
> """
> ```
> 
> I am not sure how a custom ignore pattern works here. Let me dive deeper into the whole commit linting and automation ecosystem and come back later.

This looks great. 

The custom ignore pattern hasn't been implemented in commitlint yet.
